src/Compiler/Microsoft.AspNetCore.Razor.Language/src/Components/ComponentGenericTypePass.cs

```diff
@@ -347,7 +346,7 @@ private void RewriteTypeNames(TypeNameRewriter rewriter, ComponentIntermediateNo
                     attribute.GloballyQualifiedTypeName = globallyQualifiedTypeName;
                 }
 
-                if (attribute.BoundAttribute?.IsGenericTypedProperty() ?? false && attribute.TypeName != null)
+                if ((attribute.BoundAttribute?.IsGenericTypedProperty() ?? false) && attribute.TypeName != null)
```

Comment:
```suggestion
                if (attribute.BoundAttribute?.IsGenericTypedProperty() == true && attribute.TypeName != null)
```

I know there are other instances of the original pattern, but this just seems so much more intuitive to me